crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -92,6 +92,10 @@ pub(crate) enum Instruction {
     /// Performs a function call with a list of its arguments.
     Call { func: ValueId, arguments: Vec<ValueId> },
 
+    /// Executes an "oracle" call
+    /// These are unconstrained functions that may access external state.
+    ForeignCall { func: String, arguments: Vec<ValueId> },
```

Comment:
We shouldn't need a separate instruction type for this. Instead, we can follow the example used by intrinsic functions and have a separate Value type

Reply:
I made an issue here: #1643. Good call out, I will refactor this

Reply:
It looks like there is already a separate Value type for oracles as well that we can use